src/pages/studyView/addChartButton/customCaseSelection/CustomCaseSelection.tsx

```diff
@@ -76,14 +79,26 @@ export default class CustomCaseSelection extends React.Component<ICustomCaseSele
 
     @autobind
     @action
-    onClick() {
-        let cases = this.props.selectedSamples.map(sample => {
+    onClick(selectMode: SelectMode) {
+        let selectedCases;
+        if (selectMode === SelectMode.SELECTED) {
+            selectedCases = this.props.selectedSamples;
+        } else {
+            const _selectedCaseIds = _.reduce(this.props.selectedSamples, (acc, next) => {
```

Comment:
may be `_.keyBy(this.props.selectedSamples,sample=>sample.uniqueSampleKey)`

Reply:
@kalletlak I can update. But do they have same performance? Or just for convenience?

Reply:
just a cleaner way